x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -911,17 +929,44 @@ export class ObservabilityAIAssistantClient {
     return this.dependencies.knowledgeBaseService.deleteEntry({ id });
   };
 
-  private addResponseLanguage = (messages: Message[], responseLanguage: string): Message[] => {
-    const [systemMessage, ...rest] = messages;
+  private resolveInstructions = async (
+    requestInstructions: Array<string | UserInstruction>,
+    user: { name: string },
+    namespace: string
```

Comment:
Are these arguments needed? They're available on the instance no, as dependencies?

Reply:
Sure are, I just started from the other end and didn't think further! 
Fixed in https://github.com/elastic/kibana/pull/180263/commits/932114d52857dba2c3591feeb5691b33faebb80f